lib/modules/datasource/rpm/index.ts

```diff
@@ -0,0 +1,155 @@
+import { XmlDocument } from 'xmldoc';
+import { cache } from '../../../util/cache/package/decorator';
+import { ensureTrailingSlash, joinUrlParts } from '../../../util/url';
+import { Datasource } from '../datasource';
+import type { GetReleasesConfig, Release, ReleaseResult } from '../types';
+
+export class RpmDatasource extends Datasource {
+  static readonly id = 'rpm';
+
+  // repomd.xml is a standard file name in RPM repositories which contains metadata about the repository
+  static readonly repomdXmlFileName = 'repomd.xml';
+
+  constructor() {
+    super(RpmDatasource.id);
+  }
+
+  /**
+   * Users are able to specify custom RPM repositories as long as they follow the format.
+   * There is a URI http://linux.duke.edu/metadata/filelists in the <sha>-filelists.xml.
+   * But according to this post, it's not something we can really look into or reference.
+   * @see{https://lists.rpm.org/pipermail/rpm-ecosystem/2015-October/000283.html}
+   */
+  override readonly customRegistrySupport = true;
+
+  /**
+   * Fetches the release information for a given package from the registry URL.
+   *
+   * @param registryUrl - the registryUrl should be the folder which contains repodata.xml and its corresponding file list <sha256>-filelists.xml.gz, e.g.: https://packages.microsoft.com/azurelinux/3.0/prod/cloud-native/x86_64/repodata/
+   * @param packageName - the name of the package to fetch releases for.
+   * @returns The release result if the package is found, otherwise null.
+   */
+  @cache({
+    namespace: `datasource-${RpmDatasource.id}`,
+    key: ({ packageName }: GetReleasesConfig) => packageName,
+    ttlMinutes: 1440,
+  })
+  async getReleases({
+    registryUrl,
+    packageName,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    if (!registryUrl || !packageName) {
+      return null;
+    }
+    try {
+      const filelistsXmlUrl = await this.getFilelistsXmlUrl(
+        ensureTrailingSlash(registryUrl),
+      );
+      if (!filelistsXmlUrl) {
+        return null;
+      }
+      const release = await this.getReleasesByPackageName(
+        filelistsXmlUrl,
+        packageName,
+      );
+
+      return release;
+    } catch (err) {
+      this.handleGenericErrors(err);
+    }
+  }
+
+  @cache({
+    namespace: `datasource-${RpmDatasource.id}`,
+    key: ({ registryUrl }: GetReleasesConfig) => registryUrl ?? '',
+    ttlMinutes: 1440,
+  })
+  async getFilelistsXmlUrl(registryUrl: string): Promise<string | null> {
+    const repomdUrl = joinUrlParts(
+      registryUrl,
+      RpmDatasource.repomdXmlFileName,
+    );
+    const response = await this.http.getText(repomdUrl.toString());
+    if (response.statusCode !== 200) {
+      throw new Error(
+        `Failed to fetch repomd.xml from ${repomdUrl} (${response.statusCode})`,
+      );
+    }
+
+    // parse repomd.xml using XmlDocument
+    const data = new XmlDocument(response.body);
+    const dataElements = data.childrenNamed('data');
```

Comment:
It would probably be overkill to extract XML parsing to a zod schema at this stage. But just a heads up that it _may_ come up in a review comment upstream.